tests/units/units.cpp

```diff
@@ -758,6 +758,33 @@ TEST(Units, compareScalingFactorWithNullptrAsBothParameters)
     EXPECT_EQ(0.0, libcellml::Units::scalingFactor(nullptr, nullptr));
 }
 
+TEST(Units, compareScalingFactorWithUnitWhichHasNoParent)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create();
+    u1->setName("u1");
+    u1->addUnit("apple", 0, 1.0, 1.0);
+
+    libcellml::UnitsPtr u2 = libcellml::Units::create();
+    u2->setName("u2");
+    u2->addUnit(libcellml::Units::StandardUnit::LUX, 0, 1.0, 1.0);
+
+    EXPECT_EQ(0.0, libcellml::Units::scalingFactor(u1, u2)); // Currently function not evaluating
```

Comment:
What do you mean by this?

Reply:
Ah, that's commenting left over from when I originally pushed the unit tests to show issue #473. That can be gotten rid of (if it hasn't already been done).

Reply:
Ok, will remove it (together with some other minor cleaning up).